exporter/awskinesisexporter/exporter.go

```diff
@@ -16,32 +16,68 @@ package awskinesisexporter
 
 import (
 	"context"
-	"fmt"
+	"errors"
 
-	awskinesis "github.com/signalfx/opencensus-go-exporter-kinesis"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/kinesis"
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
 	"go.opentelemetry.io/collector/consumer"
 	"go.opentelemetry.io/collector/model/pdata"
 	"go.uber.org/zap"
 
-	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awskinesisexporter/internal/translate"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awskinesisexporter/internal/batch"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awskinesisexporter/internal/producer"
 )
 
 // Exporter implements an OpenTelemetry trace exporter that exports all spans to AWS Kinesis
 type Exporter struct {
-	awskinesis *awskinesis.Exporter
-	ew         translate.ExportWriter
-	logger     *zap.Logger
+	producer producer.Batcher
+	batcher  batch.Encoder
 }
 
-var _ component.TracesExporter = (*Exporter)(nil)
+var (
+	_ component.TracesExporter  = (*Exporter)(nil)
+	_ component.MetricsExporter = (*Exporter)(nil)
+	_ component.LogsExporter    = (*Exporter)(nil)
+)
+
+func createExporter(c config.Exporter, log *zap.Logger) (*Exporter, error) {
```

Comment:
This by default creates a trace exporter given that the batching uses `NewJaeger`. What's the plan to extend with the other types and does it make more sense to name it like so and create different exporters for each.